File: javascript/problem016.js
var value = [1];

for(i=1; i<=1000; i++){
  for(j=0; j<value.length; j++){
    value[j] = value[j] * 2;
  }
  for(j=0; j<value.length; j++){
    if(value[j] >= 10){
      if(value[j+1] == undefined){
        value[j+1] = parseInt(value[j] / 10);
      }
      else{
        value[j+1] += parseInt(value[j] / 10);
      }
      value[j] = value[j] % 10;
    }
  }
}

var result = 0;
for(a=0; a<value.length; a++){
  result = result + value[a];
}
console.log(result);

File: java/src/Problem016.java
import java.math.BigInteger;

/**
 * Created by 김 on 2016-02-05.
 */
public class Problem016 {

    int result(){
        BigInteger value = new BigInteger("1");
        for(int i=1; i<=1000; i++){
            value = value.multiply(BigInteger.valueOf(2));
        }
        String str = value.toString();
        int resultValue = 0;
        for(int j=0; j<str.length(); j++){
            resultValue = resultValue + Integer.parseInt(str.substring(j,j+1));
        }
        return resultValue;
    }

    public static void main(String args[]) {
        Problem016 problem = new Problem016();
        System.out.println(problem.result());
    }
}